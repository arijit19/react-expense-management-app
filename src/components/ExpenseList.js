import React from 'react';
import {connect} from 'react-redux';

import ExpenseItem from './ExpenseItem.js';
import ExpenseSelector from '../Redux/selectors/ExpenseSelector.js';

export const ExpenseList = (props) => (
    <div className='content-container'>
        <div className='list-header'>
            <div className='show-for-mobile'>Expenses</div>
            <div className='show-for-desktop'>Expense</div>
            <div className='show-for-desktop'>Amount</div>
        </div>
        <div className='list-body'>
            {
            props.expenses.length === 0 ? (
                <div className='list-item list-item--message'>
                    <span>No Expenses</span>
                </div>
            ): props.expenses.map((expense, index)=> <ExpenseItem key={expense.id} {...expense}/>)
            }
        </div>
    </div>
);

const mapStateToProps = (state)=> {
    return {
        expenses: state.expenses ? ExpenseSelector(state.expenses,state.filters) : []
    }
}

export default connect(mapStateToProps)(ExpenseList);