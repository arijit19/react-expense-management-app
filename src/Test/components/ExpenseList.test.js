import React from 'react';
import {shallow} from 'enzyme'

import {ExpenseList} from '../../components/ExpenseList.js'
import expenses from '../fixtures/Expense.js';

test('Testing Expense list component with expense props', ()=>{
    const wrapper = shallow(<ExpenseList expenses={expenses}/>)
    expect(wrapper).toMatchSnapshot();
})

test('Testing Expense list component without expense props', ()=>{
    const wrapper = shallow(<ExpenseList expenses={[]}/>)
    expect(wrapper).toMatchSnapshot();
})