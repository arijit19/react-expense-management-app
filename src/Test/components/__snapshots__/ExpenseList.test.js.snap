// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Expense list component with expense props 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseItem
      amount={20000}
      createdAt={0}
      description="Rent"
      id="1"
      key="1"
      note=""
    />
    <ExpenseItem
      amount={1000}
      createdAt={-345600000}
      description="Game"
      id="2"
      key="2"
      note=""
    />
    <ExpenseItem
      amount={2500}
      createdAt={345600000}
      description="Internet"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`Testing Expense list component without expense props 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item--message"
    >
      <span>
        No Expenses
      </span>
    </div>
  </div>
</div>
`;
